actors/verifreg/tests/verifreg_actor_test.rs

```diff
@@ -888,6 +894,65 @@ mod allocs_claims {
         assert!(h.load_claim(&mut rt, PROVIDER1, id2).is_none()); // removed
         h.check_state(&rt);
     }
+
+    #[test]
+    fn extend_claims_restricted_correctly() {
+        let (h, mut rt) = new_harness();
+        let size = MINIMUM_VERIFIED_ALLOCATION_SIZE as u64;
+        let sector = 0;
+        let start = 0;
+        let min_term = MINIMUM_VERIFIED_ALLOCATION_TERM;
+        let max_term = min_term + 1000;
+
+        let claim1 = make_claim("1", CLIENT1, PROVIDER1, size, min_term, max_term, start, sector);
+        let claim2 = make_claim("2", CLIENT1, PROVIDER1, size, min_term, max_term, start, sector);
+        let claim3 = make_claim("3", CLIENT1, PROVIDER2, size, min_term, max_term, start, sector);
```

Comment:
Nit: drop down to a single claim so this test is more focussed on the single topic of restricting access.